src/components/layout/header/header.tsx

```diff
@@ -0,0 +1,47 @@
+import React, { ReactElement, useState, useEffect } from 'react';
+import { Layout, Menu } from 'antd';
+// import { ClickParam } from 'antd/es/menu';
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+import { faDice } from '@fortawesome/free-solid-svg-icons';
+import { Link, useLocation } from 'react-router-dom';
+import { MenuInfo } from 'rc-menu/lib/interface';
+import './styles.scss';
+
+const { Header: AntdHeader } = Layout;
+const { Item } = Menu;
+
+const Header = (): ReactElement => {
+  const [activeMenu, setActiveMenu] = useState('home');
+  const location = useLocation();
+
+  useEffect(() => {
+    const path = location.pathname.split('/');
+    setActiveMenu(path[1]);
+  }, [location]);
+
+  const onMenuItemClick = (e: MenuInfo) => {
+    setActiveMenu(e.key as string);
```

Comment:
use `e.item.key` so we don't need to cast it

Reply:
Error: Property 'key' does not exist on type 'ReactInstace'